packages/database/src/Builder/QueryBuilders/InsertQueryBuilder.php

```diff
@@ -40,6 +43,14 @@ final class InsertQueryBuilder implements BuildsQuery
 
     public ModelInspector $model;
 
+    private Database $database {
+        get => get(Database::class, $this->onDatabase);
```

Comment:
If we can avoid service location, we should. I still have to read through the rest of the code, so maybe it'll become clear why we need this, but just sharing my first thoughts up front.

I'm also ok keeping it for now, but refactoring these query builders at a later stage to support proper DI. That's outside the scope of this PR, of course. So it's not a blocker, just wanted to share my thoughts.